crossbar/_log_categories.py

```diff
@@ -0,0 +1,50 @@
+#####################################################################################
+#
+#  Copyright (C) Tavendo GmbH
+#
+#  Unless a separate license agreement exists between you and Tavendo GmbH (e.g. you
+#  have purchased a commercial license), the license terms below apply.
+#
+#  Should you enter into a separate license agreement after having received a copy of
+#  this software, then the terms of such license agreement replace the terms below at
+#  the time at which such license agreement becomes effective.
+#
+#  In case a separate license agreement ends, and such agreement ends without being
+#  replaced by another separate license agreement, the license terms below apply
+#  from the time at which said agreement ends.
+#
+#  LICENSE TERMS
+#
+#  This program is free software: you can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License, version 3, as published by the
+#  Free Software Foundation. This program is distributed in the hope that it will be
+#  useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU Affero General Public License Version 3 for more details.
+#
+#  You should have received a copy of the GNU Affero General Public license along
+#  with this program. If not, see <http://www.gnu.org/licenses/agpl-3.0.en.html>.
+#
+#####################################################################################
+
+from __future__ import absolute_import, division, print_function
+
+# The log categories
+log_categories = {
+
+    # CBXXX - Generic Crossbar logs
+    "CB500": "Unhandled exception in Crossbar.",
+
+    # ARXXX - Adapter, REST Bridge
+    "AR100": "REST bridge request recieved.",
+    "AR400": "Malformed request to the REST bridge.",
+    "AR405": "Method not accepted by the REST bridge.",
+    "AR413": "Request too long.",
+    "AR450": "Non-accepted request encoding.",
+    "AR451": "Non-decodable request body.",
+    "AR452": "Non-accepted content type.",
+}
```

Comment:
Yes -- but I was thinking more logs of any kind -- errors, warnings, successes all have codes. For example, `AR100` is "received a request" -- it's not an error, just an informational code, but it's there. This comes in really handy if you have logs going somewhere, as it's a token you can search on. For example, in Windows, to audit all successful logins, you go to the Log Viewer and make a search for the "successful authentication" log message -- it's really handy.

And yes, all things should have it. When enough do, we can start asserting on requiring one as a test suite, to make sure all our log messages are categorised, and that nothing slips through the cracks.

Currently the category doesn't show in the output -- I am thinking that maybe adding it to the start of every message in "debug mode", and to the start of all error messages in standard modes, would be a good idea? Not sure.